nomad/resource_variable.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nomad
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVariableWrite,
+		Update: resourceVariableWrite,
+		Delete: resourceVariableDelete,
+		Read:   resourceVariableRead,
+		Exists: resourceVariableExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
I think we may be able to apply some of the validation rules here:
https://github.com/hashicorp/nomad/blob/c7c8bd4ce2fcd9f2cd13278043b7baf1f8dd4f47/nomad/structs/variables.go#L189-L213

More specifically:
* the path cannot be `nomad`.
* if the path starts with `nomad/` it can only be `nomad/jobs`, `nomad/jobs/*`, or `nomad/job-templates/*`.

A description would be good as well.

Reply:
Would it make sense to use `validation.StringMatch` with a regex of questionable complexity, or a custom validation function that does some simpler validation but is larger in code.